consensus/src/quorum_store/quorum_store_wrapper.rs

```diff
@@ -249,64 +282,83 @@ impl QuorumStoreWrapper {
+                    debug!(
+                        "QS: GetBlockRequest excluded len {}, block len {}",
+                        excluded_proofs.len(),
+                        proof_block.len()
+                    );
+                    Payload::InQuorumStore(proof_block)
+                });
                 callback
                     .send(Ok(res))
-                    .expect("BlcokResponse receiver not available");
+                    .expect("BlockResponse receiver not available");
             }
             WrapperCommand::CleanRequest(logical_time, digests) => {
                 debug!("QS: got clean request from execution");
+                assert!(
+                    self.latest_logical_time < logical_time,
+                    "Non-increasing logical time"
+                );
                 self.latest_logical_time = logical_time;
+                for batch_id in self.batch_expirations.expire(logical_time.round()) {
+                    if self.batches_in_progress.remove(&batch_id).is_some() {
+                        debug!(
+                            "QS: expired batch w. id {} from batches_in_progress, new size {}",
+                            batch_id,
+                            self.batches_in_progress.len(),
+                        );
+                    }
+                }
                 for digest in digests {
-                    self.batches_to_filter.remove(&digest);
-                    self.batches_for_consensus.remove(&digest);
+                    if self.proofs_for_consensus.remove(&digest).is_some() {
+                        debug!(
+                            "QS: removed digest {} from batches_for_consensus, new size {}",
+                            digest,
+                            self.proofs_for_consensus.len(),
+                        );
+                    }
                 }
-                // TODO: remove but make CleanRequest not a ConsensusRequest,
-                // TODO: as notify_commit now calls CleanRequest.
-                // self.data_manager
-                //     .notify_commit(self.latest_logical_time)
-                //     .await;
             }
         }
     }
 
-    // TODO: use tokio select for the internal timeout feature
     pub async fn start(
         mut self,
         mut network_sender: NetworkSender,
         mut consensus_receiver: Receiver<WrapperCommand>,
         mut shutdown: Receiver<()>,
         mut network_msg_rx: aptos_channel::Receiver<PeerId, VerifiedEvent>,
     ) {
-        let mut scheduled_pulls: FuturesUnordered<ScheduledPull> = FuturesUnordered::new();
-        scheduled_pulls.push(ScheduledPull::new(
-            Instant::now() + Duration::from_millis(50),
-            false,
-        ));
         let mut proofs_in_progress: FuturesUnordered<BoxFuture<'_, _>> = FuturesUnordered::new();
 
+        // TODO: parameter? bring back back-off?
+        let mut interval = time::interval(Duration::from_millis(50));
```

Comment:
Why do you start a new Interval every iteration? 
Why don't just tick every 50 ms?